addons/hr/static/src/core/common/hr_employee_public_model.js

```diff
@@ -0,0 +1,16 @@
+import { Record, fields } from "@mail/core/common/record";
+
+export class HrEmployeePublic extends Record {
+    static _name = "hr.employee.public";
+    static id = "id";
+
+    /** @type {number} */
+    id;
+    /** @type {number} */
+    company_id = fields.One("res.company");
+    department_id = fields.One("hr.department");
+    user_id = fields.One("res.users");
+    work_location_id = fields.One("hr.work.location");
+}
```

Comment:
For now we also repeat the non-relational fields to help documentation and auto-complete.